CustomSecuritySample/README.md

```diff
@@ -0,0 +1,216 @@
+	< /CodeGroup>
+	```
+Note: 
+For simplicity, the Forms Authentication Sample is weak-named and requires a simple URL membership entry in the security policy files. In your production security extension implementation, you should create strong-named assemblies and use the strong name membership condition when adding security policies for your assembly. For more information about strong-named assemblies, see the Creating and Using Strong-Named Assemblies topic on MSDN. 
+
+To modify the Web.config file for Report Server
+-	Open the Web.config file in a text editor. By default, the file is in the <install>\ReportServer directory.
+-	Locate the <identity> element and set the Impersonate attribute to false. * <identity impersonate="false" /> * 
+-	Locate the <authentication> element and change the Mode attribute to Forms. 
+-	Add the following <forms> element as a child of the <authentication> element and set the loginUrl, name, timeout, and path attributes as follows: 
+
+	```xml
+	<authentication mode="Forms">
+		<forms loginUrl="logon.aspx" name="sqlAuthCookie" timeout="60" path="/"></forms>
+	</authentication> 
+	```
+Add the following <authorization> element directly after the <authentication> element. 
+
+	```xml
+	<authorization> 
+	<deny users="?" />
+	</authorization> 
+	```
+
+This will deny unauthenticated users the right to access the report server. The previously established loginUrl attribute of the <authentication> element will redirect unauthenticated requests to the Logon.aspx page.
+
+
+## Step 4: Some of the other changes required in the web.config file and Microsoft.ReportingServices.Portal.WebHost.exe.config
+
+Adding Machine Keys
+
+-	For the case of Forms authentication which requires the decryption of the Authentication cookie, both processes need to be configured with the same machine key and decryption algorithm. This was a step familiar to those who had previously setup SSRS to work on scale-out environments, but now is a requirement even for deployments on a single machine.
+-	For example:In <RSPATH>\ReportServer\web.config,add under "system.web"
+
+	```xml
+		<machineKey validationKey="00A2CEAEF8A91B29F63399CBEE18F272159F114991EA7CF2FD78BC5D9BB0821825C7332C4A4C1698FA58E39634365A97DA8F720377B84F471A3A166CFCDD31DF" decryptionKey="009CA6A1D48DC4DB59E54865C470DFB75FBC1B73AA4833523C9795B1FA88CBE3" validation="AES" decryption="AES" />
+	```
+-	Then <RSPATH>\RSWebApp\Microsoft.ReportingServices.Portal.WebHost.exe.config, add under "configuration"
+
+ 	```xml
+	 <system.web>
+		<machineKey validationKey="00A2CEAEF8A91B29F63399CBEE18F272159F114991EA7CF2FD78BC5D9BB0821825C7332C4A4C1698FA58E39634365A97DA8F720377B84F471A3A166CFCDD31DF" decryptionKey="009CA6A1D48DC4DB59E54865C470DFB75FBC1B73AA4833523C9795B1FA88CBE3" validation="AES" decryption="AES" />
+	</system.web>
+	```
+Note: You should use a validation key specific for you deployment, there are several tools to generate the keys, one example is: 
```

Comment:
Suggest using IIS Manager instead of an online tool for generating keys. I don't think it is a good idea to generate keys using an online tool.